utils/database_utils/mongo_utils/queries/dao/abc.py

```diff
@@ -0,0 +1,176 @@
+        **kwargs
+    ) -> dict:
+        items = {}
+        contains_filter = self._build_contains_filter(key=key, lookup_set=source_set)
+        if contains_filter:
+            filters = kwargs.pop("filters", []) + [contains_filter]
+            items = self.list_items(filters=filters, *args, **kwargs)
+            if aggregate_result:
+                items = self.aggregate_items_by_key(key=key, items=items)
+        return items
+
+    def list_items(
+        self,
+        filters: list[MongoFilter] = None,
+        projection_attributes: list = None,
+        limit: int = None,
+        result_as_cursor: bool = True,
+    ) -> dict:
+        """
+        Lists items under provided document belonging to source set of provided column values
+
+        :param filters: filters to consider (optional)
+        :param projection_attributes: list of value keys to return (optional)
+        :param limit: limit number of returned attributes (optional)
+        :param result_as_cursor: to return result as cursor (defaults to True)
+        :returns results of FIND operation over the desired document according to applied filters
+        """
+        result_filters = {}
+        if limit:
+            result_filters["limit"] = limit
+        items = self._execute_query(
+            command=MongoCommands.FIND_ALL,
+            filters=filters,
+            result_filters=result_filters,
+            result_as_cursor=result_as_cursor,
+        )
+        # TODO: pymongo support projection only as aggregation API which is not yet implemented in project
+        if projection_attributes:
+            items = [
+                {k: v}
+                for item in items
+                for k, v in item.items()
+                if k in projection_attributes
+            ]
+        return items
+
+    def aggregate_items_by_key(self, key, items: list) -> dict:
+        aggregated_data = {}
+        # TODO: consider Mongo DB aggregation API
+        for item in items:
+            items_key = item.pop(key, None)
+            if items_key:
+                aggregated_data.setdefault(items_key, []).append(item)
+        return aggregated_data
+
+    def _build_list_items_filter(
+        self, key, lookup_set, additional_filters: list[MongoFilter]
+    ) -> list[MongoFilter] | None:
+        mongo_filters = additional_filters or []
+        contains_filter = self._build_contains_filter(key=key, lookup_set=lookup_set)
+        if contains_filter:
+            mongo_filters.append(contains_filter)
+        return mongo_filters
+
+    def _build_contains_filter(self, key, lookup_set) -> MongoFilter | None:
+        mongo_filter = None
+        if key and lookup_set:
+            lookup_set = list(set(lookup_set))
+            mongo_filter = MongoFilter(
+                key=key,
+                value=lookup_set,
+                logical_operator=MongoLogicalOperators.IN,
+            )
+        return mongo_filter
+
+    def add_item(self, data: dict):
```

Comment:
missing return annotation and docstring

Reply:
> missing return annotation and docstring

actually return value is never used, I just use return for the visualization of the function callback